### ISSUE TITLE
Extension conditions should perform object picking (like normal event conditions)

### ISSUE THREAD

**Issue by tristanbob** [4ian/GDevelop#2817]
## Description

Extension conditions currently only provide a true/false functionality.  Unlike normal event conditions, extension conditions DO NOT perform object picking.  This means that users are required to always use a ForEach() loop when using extension conditions on multiple instances. 
 
Referenced in these discussions: 

https://forum.gdevelop-app.com/t/do-extension-conditions-perform-object-picking/33811/16

https://github.com/GDevelopApp/GDevelop-extensions/issues/124

## Solution suggested

Extension conditions should perform object picking the same way that normal event conditions do.  This will provide the most consistent experience for users.

**Comment by D8H** [comment]
There is a use case and a workaround with a JavaScript event: https://github.com/GDevelopApp/GDevelop-extensions/issues/124

**Comment by tristanbob** [comment]
Is it also true that extension actions that create objects do not perform object picking?  (normal object creation adds that instance to object picking)

![image](https://user-images.githubusercontent.com/8879811/132268765-b7786e8e-064b-4f9c-a4bb-be503a1dcc76.png)

I was expecting that object timer to apply to the WhitePixel objects that were created in that action above, but instead, it is applying to ALL instances

**Comment by D8H** [comment]
Yes, I like the Arthuro's solution: https://github.com/GDevelopApp/GDevelop-extensions/issues/124#issuecomment-823269197

**Comment by D8H** [comment]
Related PR:
- https://github.com/4ian/GDevelop/pull/3163
